File: Problems/leetcode/1. Two Sum/two_sum.java
/* TC- O(N)
SC - O(N) */
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int a[]= new int[2];
        int index=0;
        for(index=0;index<nums.length;index++){
            
                if(map.containsKey(target-nums[index]) && (target-nums[index]!=nums[index] ||
                   map.get(Math.abs(target-nums[index]))!=index))
                {
                    
                   a[0]= map.get(target-nums[index]);
                   a[1]= index;
                   
                   return(a);
                      
                }
                else{
                    map.put(nums[index],index);
                }
         
        }
        return null;
    }
        
    }

File: Problems/leetcode/1. Two Sum/two_sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
     map<int,int> cache ; 
     vector<int> output;
     int B=0;
        for(int i=0;i<nums.size();i++){
            B=target-nums[i]; 
            if(cache.count(B)>0 and (B!=nums[i] or cache[B]!=i)){
                output.push_back(i);
                output.push_back(cache[B]);
                return output;
            }
            else
            {
                cache[nums[i]]=i;
            }
        }
         return output;
        
    }
};